tao_compiler/mlir/disc/transforms/disc_transpose_simplifier.cc

```diff
@@ -687,15 +687,15 @@ LogicalResult reverseIfOperandsAreConsistentTransposeOps(Operation* op,
 
   OpBuilder b(op);
   SmallVector<Value> newOperands;
-  newOperands.push_back(insertTranspose(op, lhs, permLHS, b)->getResult(0));
-  newOperands.push_back(insertTranspose(op, rhs, permLHS, b)->getResult(0));
+  SmallVector<int64_t> reversePerm = getReversePermutation(permLHS);
```

Comment:
So the bug origins from two consecutive TransposeOp what could not get eliminated?

Reply:
Yes. The permutation is wrong. Original UT is a special case where `permLHS (permLHS) = identity`, thus it's ok even the permutation is wrong. I added a new UT.